lib/mongoid/association/embedded/embeds_one/binding.rb

```diff
@@ -21,7 +21,10 @@ class Binding
           def bind_one
             _target.parentize(_base)
             binding do
-              _target.do_or_do_not(_association.inverse_setter(_target), _base)
+              begin
+                _target.do_or_do_not(_association.inverse_setter(_target), _base)
+              rescue Mongoid::Errors::AmbiguousRelationship => exception
```

Comment:
Can you please add a comment here for future reference summarizing our findings and specifically giving an explanation 1) when this exception would be raised and 2) why ignoring this exception does the right thing?

The exception also doesn't need to be assigned to the variable since you aren't doing anything with it subsequently.

Reply:
I actually removed this piece of code, put the characterization before the binding and add a check for multiple inverses to the binding.